plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/ElkSpacings.java

```diff
@@ -0,0 +1,302 @@
+         * values when setting the spacing value for a concrete node.
+         */
+        private Map<IProperty<Double>, Double> factorMap = Maps.newHashMap();
+                
+        /**
+         * Constructs a new builder and computes the initial factors based on the passed based value.
+         */
+        protected AbstractSpacingsBuilder(final double theBaseSpacing) {
+            this.baseSpacing = theBaseSpacing;
+
+            // Put the base spacing option in the map as well to allow iterating all options by using this map
+            factorMap.put(getBaseSpacingOption(), 1.0d);
+
+            // Initially, and by default, set all factors to the ratio given by default values set in the Core.melk
+            getDependendSpacingOptions().forEach(option -> {
+                double factor = (getBaseSpacingOption().getDefault() != null && option.getDefault() != null)
+                        ? option.getDefault() / getBaseSpacingOption().getDefault()
+                        : 1.0d;
+                factorMap.put(option, factor);
+            });
+        }
+        
+        // Protected
+
+        /**
+         * See {@link #baseSpacing}'s javadoc for further details.
+         * 
+         * @return a spacing option that shall serve as the basis for the configuration, i.e. it will serve as basis to
+         *         compute factors to be applied to other spacing values.
+         */
+        protected abstract IProperty<Double> getBaseSpacingOption();
+
+        /**
+         * @return a list of spacing options that shall be affected by the specified base value.
+         */
+        protected abstract Iterable<IProperty<Double>> getDependendSpacingOptions();
+        
+        /**
+         * @return the filters that shall be applied during configuration.
+         */
+        protected List<IPropertyHolderOptionFilter> getFilters() {
+            return filters;
+        }
+
+        /**
+         * @return type of the implementing class - necessary to make suppress a warning whenever a method in this base
+         *         class returns {@code T}.
+         */
+        protected abstract T thisT();
+        
+        /**
+         * @return the factor to be used for {@code value} relative to the specified {@link #baseSpacing}.
+         */
+        protected double computeFactor(final double value) {
+            return value / baseSpacing;
+        }
+        
+        // Public API
+        
+        /**
+         * @return an immutable list of the currently configured factors.
+         */
+        public Map<IProperty<Double>, Double> getFactors() {
+            return ImmutableMap.copyOf(factorMap);
+        }
+
+        /**
+         * Configure the {@code factor} to use for {@code spacingOption} relative to the configured base value.
+         * 
+         * @return this instance.
+         */
+        public T withFactor(final IProperty<Double> spacingOption, final double factor) {
```

Comment:
Would it be sensible to throw exceptions for negative factors (and values, for that matter)?

Reply:
Good point.